src/main/java/org/cdc/gov/sdp/aphl/AphlS3Endpoint.java

```diff
@@ -0,0 +1,145 @@
+ * S3 Storage Service.
+ */
+@UriEndpoint(producerOnly = true, scheme = "aphl-s3", title = "APHL Push only AWS S3 Storage Service", syntax = "aphl-s3:bucketNameOrArn", label = "cloud,file")
+public class AphlS3Endpoint extends DefaultEndpoint {
+
+	private static final Logger LOG = LoggerFactory.getLogger(S3Endpoint.class);
+
+	private AmazonS3 s3Client;
+
+	@UriPath(description = "Bucket name or ARN")
+	@Metadata(required = "true")
+	private String bucketNameOrArn; // to support component docs
+	@UriParam
+	private S3Configuration configuration;
+	@UriParam(label = "consumer", defaultValue = "10")
+	private int maxMessagesPerPoll = 10;
+
+	@Deprecated
+	public AphlS3Endpoint(String uri, CamelContext context, S3Configuration configuration) {
+		super(uri, context);
+		this.configuration = configuration;
+		ensureClient();
+	}
+
+	public AphlS3Endpoint(String uri, Component comp, S3Configuration configuration) {
+		super(uri, comp);
+		this.configuration = configuration;
+		ensureClient();
+	}
+
+	public Consumer createConsumer(Processor processor) throws Exception {
+		return null;
+	}
+
+	public Producer createProducer() throws Exception {
+		return new AphlS3Producer(this);
+	}
+
+	public boolean isSingleton() {
+		return true;
+	}
+
+	public void ensureClient() {
+
+		s3Client = configuration.getAmazonS3Client() != null ? configuration.getAmazonS3Client() : createS3Client();
+
+		if (ObjectHelper.isNotEmpty(configuration.getAmazonS3Endpoint())) {
+			s3Client.setEndpoint(configuration.getAmazonS3Endpoint());
+		}
+		
+		LOG.debug("started CDC APHL Producer only AWS S3 client");
+	}
+
+	public S3Configuration getConfiguration() {
+		return configuration;
+	}
+
+	public void setConfiguration(S3Configuration configuration) {
+		this.configuration = configuration;
+	}
+
+	public void setS3Client(AmazonS3 s3Client) {
+		this.s3Client = s3Client;
+	}
+
+	public AmazonS3 getS3Client() {
+		return s3Client;
+	}
+
+	/**
+	 * Provide the possibility to override this method for an mock
+	 * implementation
+	 */
+	AmazonS3 createS3Client() {
+
+		AmazonS3Client client = null;
+		ClientConfiguration clientConfiguration = null;
+		boolean isClientConfigFound = false;
+		if (hasProxyConfiguration()) {
+			clientConfiguration = new ClientConfiguration();
+			clientConfiguration.setProxyHost(configuration.getProxyHost());
+			clientConfiguration.setProxyPort(configuration.getProxyPort());
+			isClientConfigFound = true;
+		}
+		LOG.info("AWS ACCESS KEY "+configuration.getAccessKey());
+		LOG.info("AWS SECRET KEY "+configuration.getSecretKey());
+		if (configuration.getAccessKey() != null && configuration.getSecretKey() != null) {
+			AWSCredentials credentials = new BasicAWSCredentials(configuration.getAccessKey(),
+					configuration.getSecretKey());
+			if (isClientConfigFound) {
+				client = new AmazonS3Client(credentials, clientConfiguration);
+			} else {
+				client = new AmazonS3Client(credentials);
+			}
+		} else {
+			if (isClientConfigFound) {
```

Comment:
This class was cobbled together from the s3 class in camel.  I believe what they are trying to do is determine whether or not to use override values for the credentials instead of what could be found in the client config.